x-pack/plugins/ml/public/application/components/import_export_jobs/export_jobs_flyout/export_jobs_flyout.tsx

```diff
@@ -170,26 +171,33 @@ export const ExportJobsFlyout: FC<Props> = ({ isDisabled, currentTab }) => {
     }
   }
 
-  const attemptTabSwitch = useCallback(() => {
-    // if the user has already selected some jobs, open a confirm modal
-    // rather than changing tabs
-    if (selectedJobIds.length > 0) {
-      setSwitchTabConfirmVisible(true);
-      return;
-    }
+  const attemptTabSwitch = useCallback(
+    (jobType: JobType) => {
+      if (jobType === selectedJobType) {
+        return;
+      }
+      // if the user has already selected some jobs, open a confirm modal
+      // rather than changing tabs
+      if (selectedJobIds.length > 0) {
+        setSwitchTabNextTab(jobType);
+        setSwitchTabConfirmVisible(true);
+        return;
+      }
 
-    switchTab();
-  }, [selectedJobIds]);
+      switchTab(jobType);
+    },
+    [selectedJobIds]
+  );
 
   useEffect(() => {
     setSelectedJobDependencies(
       jobDependencies.filter(({ jobId }) => selectedJobIds.includes(jobId))
     );
   }, [selectedJobIds]);
 
-  function switchTab() {
-    const jobType =
-      selectedJobType === 'anomaly-detector' ? 'data-frame-analytics' : 'anomaly-detector';
+  function switchTab(jobType: JobType) {
+    // const jobType =
+    //   selectedJobType === 'anomaly-detector' ? 'data-frame-analytics' : 'anomaly-detector';
```

Comment:
Should this comment stay or can it be removed?

Reply:
Updated in https://github.com/elastic/kibana/pull/110578/commits/538cb1b3751088eade0a21dab325097873bc50f9